synapse/api/errors.py

```diff
@@ -682,18 +682,27 @@ class FederationPullAttemptBackoffError(RuntimeError):
     Attributes:
         event_id: The event_id which we are refusing to pull
         message: A custom error message that gives more context
+        retry_after_ms: The remaining backoff interval, in milliseconds
     """
 
-    def __init__(self, event_ids: List[str], message: Optional[str]):
-        self.event_ids = event_ids
+    def __init__(
+        self, event_ids: Iterable[str], message: Optional[str], retry_after_ms: int
```

Comment:
Drive-by comment:

```suggestion
        self, event_ids: StrCollection, message: Optional[str], retry_after_ms: int
```

Reply:
Good idea, thanks!

Reply:
Context for why: https://github.com/matrix-org/synapse/issues/14809